ocrd_network/ocrd_network/rabbitmq_utils/ocrd_messages.py

```diff
@@ -0,0 +1,120 @@
+from __future__ import annotations
+from datetime import datetime
+from typing import Any, Dict, List, Optional
+import yaml
+from ocrd_network.models.job import Job
+
+
+class OcrdProcessingMessage:
+    def __init__(
+            self,
+            job_id: str = None,
+            processor_name: str = None,
+            created_time: int = None,
+            path_to_mets: str = None,
+            workspace_id: str = None,
+            input_file_grps: List[str] = None,
+            output_file_grps: Optional[List[str]] = None,
+            page_id: str = None,
+            parameters: Dict[str, Any] = None,
+            result_queue_name: str = None,
+            callback_url: str = None
+    ) -> None:
+        if not job_id:
+            raise ValueError('job_id must be set')
+        if not processor_name:
+            raise ValueError('processor_name must be set')
+        if not created_time:
+            # We should not raise a ValueError but just calculate it
+            created_time = int(datetime.utcnow().timestamp())
+        if not input_file_grps or len(input_file_grps) == 0:
+            raise ValueError('input_file_grps must be set and contain at least 1 element')
+        if not (workspace_id or path_to_mets):
+            raise ValueError('Either `workspace_id` or `path_to_mets` must be set')
+
+        self.job_id = job_id  # uuid
+        self.processor_name = processor_name  # "ocrd-.*"
+        # Either of these two below
+        self.workspace_id = workspace_id  # uuid
+        self.path_to_mets = path_to_mets  # absolute path
+        self.input_file_grps = input_file_grps
+        self.output_file_grps = output_file_grps
+        # e.g., "PHYS_0005..PHYS_0010" will process only pages between 5-10
+        self.page_id = page_id if page_id else None
+        # processor parameters
+        self.parameters = parameters if parameters else None
+        self.result_queue_name = result_queue_name
+        self.callback_url = callback_url
+        self.created_time = created_time
+
+    @staticmethod
+    def encode_yml(ocrd_processing_message: OcrdProcessingMessage) -> bytes:
+        """convert OcrdProcessingMessage to yml
+        """
+        return yaml.dump(ocrd_processing_message.__dict__, indent=2).encode('utf-8')
+
+    @staticmethod
+    def decode_yml(ocrd_processing_message: bytes) -> OcrdProcessingMessage:
+        """Parse OcrdProcessingMessage from yml
+        """
+        msg = ocrd_processing_message.decode('utf-8')
+        data = yaml.load(msg, Loader=yaml.Loader)
```

Comment:
- Please use `yaml.safe_load` instead of `load`.
- After this step, `data` should be validated against the schema. Please check how the `OcrdToolValidator` in `ocrd_validators/ocrd_tool_validator.py` is implemented and do it in the same way.

Reply:
You know this is problematic, right? Some fields of the processing message have to be `None` but the schema does not accept that. It either accepts `str` or just a missing key field. Anyway, resolved although an extra workaround was needed inside the `process_message` method.